network-formula/network/systemd/link.sls

```diff
@@ -0,0 +1,53 @@
+{#-
+Salt state file for managing network devices using systemd
+Copyright (C) 2025 Georg Pfuetzenreuter <mail+opensuse@georg-pfuetzenreuter.net>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+-#}
+
+{%- set mypillar = salt['pillar.get']('network:systemd', {}) %}
+{%- set directory = '/etc/systemd/network/' %}
+{%- set files = [] %}
+
+{%- if mypillar %}
+network_systemd_link_config:
+  file.managed:
+    - names:
+        {%- for name, config in mypillar.get('link', {}).items() %}
```

Comment:
Asssuming someone has the following pillar:

```
network:
  systemd:
    clean: True
    # note: no "link" section
```

you will get an empty `- names:`. I wouldn't be too surprised if this triggers a syntax error.

Of course this could be seen as a safety net to avoid accidentally deleting all *.link files ;-) but if you want that, better make it an explicit check.

For this section, I'd recommend to change line +23 to `{%- if mypillar.get('link') %}`